src/cdk/a11y/aria-describer.ts

```diff
@@ -0,0 +1,135 @@
+   */
+  addDescription(hostElement: HTMLElement, message: string) {
+    if (!this._platform.isBrowser || !message.trim()) { return; }
+
+    if (!registeredMessages.get(message)) {
+      const messageElement = this._createMessageElement(message);
+      registeredMessages.set(message, {messageElement, hostElements: []});
+    }
+
+    const registeredMessage = registeredMessages.get(message)!;
+    registeredMessage.hostElements.push(hostElement);
+    addAriaReferencedId(hostElement, registeredMessage.messageElement.id, 'aria-describedby');
+  }
+
+  /**
+   * Removes the host element's aria-describedby reference to the message element.
+   */
+  removeDescription(hostElement: HTMLElement, message: string) {
+    if (!this._platform.isBrowser || !message.trim()) { return; }
+
+    const registeredMessage = registeredMessages.get(message)!;
+    registeredMessage.hostElements = registeredMessage.hostElements.filter(el => el != hostElement);
+    removeAriaReferencedId(hostElement, registeredMessage.messageElement.id, 'aria-describedby');
+
+    if (registeredMessage.hostElements.length == 0) {
+      this._deleteMessageElement(message);
+    }
+
+    if (messagesContainer!.childNodes.length == 0) {
+      this._deleteMessagesContainer();
+    }
+  }
+
+  /** Unregisters all created message elements and removes the message container. */
+  _unregisterAllMessages() {
+    registeredMessages.forEach((registeredMessage, message) => {
+      registeredMessage.hostElements.forEach(hostElement => {
+        this.removeDescription(hostElement, message);
+      });
+    });
+
+    registeredMessages.clear();
+  }
+
+  /**
+   * Creates a new element in the visually hidden message container element with the message
+   * as its content.
+   */
+  private _createMessageElement(message: string): HTMLElement {
+    const messageElement = this._renderer.createElement('div');
+    this._renderer.setAttribute(messageElement, 'id', `cdk-aria-describedby-message-${nextId++}`);
+    this._renderer.appendChild(messageElement, this._renderer.createText(message));
+
+    if (!messagesContainer) { this._createMessagesContainer(); }
+    this._renderer.appendChild(messagesContainer, messageElement);
+
+    return messageElement;
+  }
+
+  /** Deletes the message element from the global messages container. */
+  private _deleteMessageElement(message: string) {
+    const messageElement = registeredMessages.get(message)!.messageElement;
+    this._renderer.removeChild(messagesContainer, messageElement);
+    registeredMessages.delete(message);
+  }
+
+  /** Creates the global container for all aria-describedby messages. */
+  private _createMessagesContainer() {
+    messagesContainer = this._renderer.createElement('div');
+
+    this._renderer.setAttribute(messagesContainer, 'id', MESSAGES_CONTAINER_ID);
+    this._renderer.setAttribute(messagesContainer, 'aria-hidden', 'true');
+    this._renderer.addClass(messagesContainer, 'cdk-visually-hidden');
```

Comment:
Can this just be `display: none`? IIRC, elements w/ `display: none` can still act as labels / descriptions for `aria-labelledby` and `aria-describedby`